kernel/packages/decentraland-ecs/src/decentraland/math/Quaternion.ts

```diff
@@ -214,24 +214,25 @@ export class Quaternion {
    * @param pitch - defines the rotation around X axis
    * @param roll - defines the rotation around Z axis
    * @param result - defines the target quaternion
+   *
+   * Implemented unity-based calculations from: https://stackoverflow.com/a/56055813
    */
   public static RotationYawPitchRollToRef(yaw: number, pitch: number, roll: number, result: Quaternion): void {
-    let halfRoll = roll * 0.5
     let halfPitch = pitch * 0.5
     let halfYaw = yaw * 0.5
+    let halfRoll = roll * 0.5
 
-    let c1 = Math.cos(halfPitch)
-    let c2 = Math.cos(halfYaw)
-    let c3 = Math.cos(halfRoll)
-
-    let s1 = Math.sin(halfPitch)
-    let s2 = Math.sin(halfYaw)
-    let s3 = Math.sin(halfRoll)
+    const c1 = Math.cos(halfPitch)
+    const c2 = Math.cos(halfYaw)
+    const c3 = Math.cos(halfRoll)
+    const s1 = Math.sin(halfPitch)
+    const s2 = Math.sin(halfYaw)
+    const s3 = Math.sin(halfRoll)
 
-    result.x = s1 * c2 * c3 + c1 * s2 * s3
-    result.y = c1 * s2 * c3 - s1 * c2 * s3
-    result.z = c1 * c2 * s3 + s1 * s2 * c3
-    result.w = c1 * c2 * c3 - s1 * s2 * s3
+    result.x = c2 * s1 * c3 + s2 * c1 * s3
+    result.y = s2 * c1 * c3 - c2 * s1 * s3
+    result.z = c2 * c1 * s3 - s2 * s1 * c3
+    result.w = c2 * c1 * c3 + s2 * s1 * s3
```

Comment:
This is the exact same calculation with the operands of the multiplications flipped. The only thing that chanes are the z and w componets addition and substraction which are swapped

Reply:
Yes i know, but it is exactly as used in the referenced calculations, I thought it would be less confusing to leave it that way.